docs/external-db.md

```diff
@@ -0,0 +1,159 @@
+# External DB integration
+
+Backstage hosts the data in the [PostgreSQL database](https://backstage.io/docs/getting-started/config/database/).
+By default, Operator creates and manages local instance of PostgreSQL in the same as Backstage deployment namespace but it also allows to switch this off and configure external instance instead.
+Usually, external connection requires more security, so, this instruction includes steps to configure SSL/TLS.
+
+### Configure your external PostgreSQL instance
+As a prerequisite you have to know:
+- **db-host** - your PostgreSQL instance DNS or IP address 
+- **db-port** - your PostgreSQL instance port number (usually 5432)
+- **username** - to connect to your PostgreSQL instance
+- **password** - to connect to your PostgreSQL instance
+
+In addition, to get your database connection secured with SSL/TLS you also need certificates in a form of PEM file. 
+
+You can find configuration guidelines for:
+- [AWS RDS PostgreSQL](#aws-rds-postgresql)
+- [Azure Database PostgreSQL](#azure-db-postgresql)
+
+### Create secret with PostgreSQL connection properties:
+````
+cat <<EOF | kubectl -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <cred-secret-name>
+type: Opaque
+stringData:
+ POSTGRES_PASSWORD: <password>
+ POSTGRES_PORT: <db-port>
+ POSTGRES_USER: <username>
+ POSTGRES_HOST: <db-host>
+ PGSSLMODE: require #  for TLS connection
+ NODE_EXTRA_CA_CERTS: <abs-path-to-pem-file> # for TLS connection, e.g. /opt/app-root/src/postgres-crt.pem
+EOF
+````
+
+### Create secret with certificate(s):
+(omit this step if you do not need TLS connection, maybe for testing purpose)
+
+````
+cat <<EOF | kubectl -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <crt-secret>
+type: Opaque
+stringData:
+ postgres-crt.pem: |-
```

Comment:
Rather than copy-pasting the content of the file, how about using a command like `kubectl create secret generic NAME --from-file=...` ([ref](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_create/kubectl_create_secret_generic/)) to create a secret from a file? I think it is more user-friendly.

Reply:
Not sure I get your "more user friendly" point, of course there are more than one way to do it but IMO it is more convenient to copy/paste/change one command than do the same but with one file AND one command... matter of taste :)

Reply:
I mean, a command like this:
```shell
kubectl create secret generic <crt-secret> --from-file=postgres-crt.pem=/path/to/pem/file
```

is more convenient than the current `cat <<EOF | kubectl -n <your-namespace> create -f -` command, which requires you to paste the content of your cert file in the `stringData` field (and make sure to indent correctly)

But anyway, I'm fine with what is here as well.